File: 349-Intersection of Two Arrays/Solution.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        dict = {}
        result = []
        for i in range(len(nums1)):
            dict[nums1[i]] = 1
        for j in range(len(nums2)):
            if dict.has_key(nums2[j]):
                dict.pop(nums2[j])
                result.append(nums2[j])
        return result
                

File: 349-Intersection of Two Arrays/Solution.java
public class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set = new HashSet<>();
        Set<Integer> intersect = new HashSet<>();
        
        for(int i=0;i<nums1.length;i++){
            set.add(nums1[i]);
        }
        for (int j=0;j<nums2.length;j++){
            if(set.contains(nums2[j])){
                intersect.add(nums2[j]);
            }
        }
        int[] result = new int[intersect.size()];
        int i = 0;
        for (int num:intersect){
            result[i++] = num;
        }
        return result;
    }
}